ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java

```diff
@@ -1230,14 +1255,22 @@ public void dropTable(String dbName, String tableName, boolean deleteData,
    */
   public void dropTable(String dbName, String tableName, boolean deleteData,
       boolean ignoreUnknownTab, boolean ifPurge) throws HiveException {
+
     try {
+      Table tbl = null;
+      try {
+        tbl = getTable(dbName, tableName);
+      } catch (InvalidTableException e) {
+      }
+
```

Comment:
Why is this change needed?

Reply:
Rebased with master and only kept relevant changes.